_posts/2019-04-24-istio-service-mesh-s2s.md

```diff
@@ -0,0 +1,431 @@
+These subject can be namespace and/or specific services.
+Specifying the namespace `with-istio` in the inclusion section, enables RBAC for all services in that namespace.
+By default the RBAC configuration will reject all requests which don't have the proper access defined with a HTTP 403 error code. 
+
+```yaml
+apiVersion: "rbac.istio.io/v1alpha1"
+kind: ClusterRbacConfig
+metadata:
+  name: default
+spec:
+  mode: 'ON_WITH_INCLUSION'
+  inclusion:
+    namespaces: ['with-istio']
+```
+
+After this RBAC config is applied, requests to the test-app services will start failing again. 
+The test-app currently doesn't have a role attached to its service account that allows it to access the CouchDB database. 
+Therefor all requests to the service will be rejected with an HTTP error code of 403.
+This is shown in the following drawing.
+
+<img class="image fit" style="margin:0px auto; max-width: 417px;" src="{{ '/img/2019-04-14-istio-service-mesh-s2s/istio-rbac-refused.png' | prepend: site.baseurl }}" alt="RBAC refuses connection" />
+
+The following manifest creates a role that allows access to the CouchDB service for all GET requests on any given path.
+Note that the full service name is used in the services specification, this is currently required by Istio.
+This is only needed for Istio to identify the traffic, short names can still be used to access the service.
+By applying this service role, nothing will change to the requests to the test-app since the role is not yet connected to the service account of the test-app service. 
+
+```yaml
+apiVersion: "rbac.istio.io/v1alpha1"
+kind: ServiceRole
+metadata:
+  name: couchdb-role
+  namespace: with-istio
+spec:
+  rules:
+  - services: ["couchdb.with-istio.svc.cluster.local"]
+    methods: ["GET"]
+    paths: ['*']
+```
+
+So next, we link the new role to the service account of the test-app service.
+This is done through a service role binding.
+There are two sections to this binding: the role and the subjects.
+The role is the one that was created using the previous manifest. 
+The subjects can be any identity known to Istio. 
+In the demo scenario, only the service accounts are known.
+Istio defines a service account as a user identity. 
+As with the services names, the service account reference needs to be the full reference scoped towards the cluster.
+This allows services from outside of the namespaces to be specified as well.
+
+```yaml
+apiVersion: "rbac.istio.io/v1alpha1"
+kind: ServiceRoleBinding
+metadata:
+  name: bind-test-app-service-couchdb-role
+  namespace: with-istio
+spec:
+  subjects:
+  - user: "cluster.local/ns/with-istio/sa/test-app-test-app-chart"
+  roleRef:
+    kind: ServiceRole
+    name: "couchdb-role"
+```
+
+After applying the last manifest, requests should again be authorized and allowed to CouchDB.
+
+## Conclusion
+
+This demo showed how Istio can be used to enable strong authentication and authorization for services in a Kubernetes cluster. 
```

Comment:
This only showcases the authentication between pods via mTLS and the authorization of the communication between those two pods. It doesn't showcase the authorization between two services on application level.

Reply:
Updated, please review ;)